src/NLog/Config/XmlLoggingConfiguration.cs

```diff
@@ -321,27 +325,24 @@ private void Initialize(XmlReader reader, string fileName, bool ignoreErrors)
             catch (Exception exception)
             {
                 InitializeSucceeded = false;
-                if (exception.MustBeRethrown())
+
+                if (exception.IsServereException())
                 {
                     throw;
                 }
 
-                NLogConfigurationException ConfigException = new NLogConfigurationException("Exception occurred when loading configuration from " + fileName, exception);
+                var message = string.Format("Exception occurred when loading configuration from '{0}': {1}", fileName, exception);
 
-                if (!ignoreErrors)
+                if (ignoreErrors)
                 {
-                    if (LogManager.ThrowExceptions)
-                    {
-                        throw ConfigException;
-                    }
-                    else
-                    {
-                        InternalLogger.Error("Error in Parsing Configuration File. Exception : {0}", ConfigException);
-                    }
+                    InternalLogger.Error(message);
                 }
                 else
                 {
-                    InternalLogger.Error("Error in Parsing Configuration File. Exception : {0}", ConfigException);
+                    if (exception.MustBeRethrown(message))
+                    {
+                        throw new NLogConfigurationException(message);
+                    }
                 }
             }
         }
```

Comment:
> This case is tricky and it is forcing us to duplicate some of the exception-handling logic from MustBeRethrown() (how we log exceptions in particular). How about adding more parameters to HandleException() to accommodate this?

Well funny point. We tried that before, and I just removed it. Because:
- The were too many parameters to `MustBeRethrown`
- The code of `MustBeRethrown` was too difficult
- It's only needed in some cases.

> If we go with HandleException() I would suggest the method be void and it does the logging and then rethrows the exception itself.

We tried that also before :) But the stacktrace will be different :(  and that's a breaking change.